### PR TITLE
Update: PNG export uses dashboard layout.

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

PNG currently renders dashboards like PDFs, a long list of widgets.  This update will use the dashboard layout so that users have control of widget placement in their PNG exports.

## Proposed Changes

- Add fileType query param to print routes. This adds a class for easy conditional css styling and conditional rendering.
- Dashboard now loads gridstack and uses widget layout in print with fileType of png.

## Screenshots
export

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/dashboards/tests/acceptance/print-dashboard-test.js [hunk 1]
```diff
@@ -46,4 +51,34 @@ module('Acceptance | print dashboard', function(hooks) {
...
+    assert.expect(11);
+    await visit('/print/dashboards/1?fileType=png');
+    assert.equal(currentURL(), '/print/dashboards/1/view?fileType=png', 'Redirect to view sub route');
+
+    assert.dom('.navi-dashboard--print-png').exists('Correct filetype classes are present');
+
+    assert.dom('.grid-stack').exists('Grid stack exists in PNG view');
+
+    assert.deepEqual(
+      [...document.querySelectorAll('.navi-widget__title')].map(el => el.textContent.trim()),
```
Comment (kevinhinterlong) [inline packages/dashboards/tests/acceptance/print-dashboard-test.js:65]: findAll

File: packages/dashboards/tests/acceptance/print-dashboard-test.js [hunk 2]
```diff
@@ -46,4 +51,34 @@ module('Acceptance | print dashboard', function(hooks) {
...
+
+    assert.dom('.goal-gauge-widget').isVisible('Goal gauge shows up');
+    assert.dom('.c3-chart-arcs-title .value-title').hasText('570.53', 'Goal gauge shows correct value');
+    assert.dom('.c3-chart-arcs-title .metric-title').hasText('Ad Clicks', 'Goal gauge shows correct value');
+
+    assert.dom('.line-chart-widget').isVisible('Line chart shows up');
+    assert.deepEqual(
+      [...document.querySelectorAll('.line-chart-widget .c3-legend-item')].map(el => el.textContent),
+      ['Ad Clicks', 'Nav Link Clicks'],
+      'The legends fill in with widget dimensions'
```
Comment (kevinhinterlong) [inline packages/dashboards/tests/acceptance/print-dashboard-test.js:78]:
```suggestion
      'The legends fill in with widget metrics'
```

**Review by kevinhinterlong — APPROVED** [review]

File: packages/dashboards/addon/templates/dashboards-print/dashboards/dashboard/view.hbs [hunk 3]
```diff
@@ -1,24 +1,45 @@
...
   <div class="navi-dashboard__body">
-    {{#if @model.dashboard.widgets.length}}
-      <div class="navi-dashboard__widgets">
+    {{#if (not @model.dashboard.widgets.length)}}
+      <div class="error-container">
+        <div class="error">
+          <p>Looks like this dashboard has no widgets.</p>
+        </div>
+      </div>
+    {{else if (eq this.fileType 'png')}}
```
Comment (jkusa) [inline packages/dashboards/addon/templates/dashboards-print/dashboards/dashboard/view.hbs:10]: Could we have a layout abstraction independent of the `fileType`? Something like:
```suggestion
    {{else unless this.singleColumn}}
```

where `singleColumn` or `layout/format` is a query param.

Comment (cythrawll) [reply packages/dashboards/addon/templates/dashboards-print/dashboards/dashboard/view.hbs]: Updated query param to be 'layout' and went from there.

Removed report part as this new concept does not apply.

**Review by jkusa — APPROVED** [review]

**Pull request merged**